File: ObjcExample/TabBarController.m
//
//  TabBarController.m
//  ObjcExample
//
//  Created by Timur Ganiev on 08.12.2020.
//  Copyright © 2020 Timur Ganiev. All rights reserved.
//

#import "TabBarController.h"

@implementation TabBarController

#pragma mark -  Public

#pragma mark Overrides

- (void)viewDidLoad {

    [super viewDidLoad];

    // Make the bottom tab bar translucent

    self.horizontalTabBar.backgroundColor = [UIColor clearColor];
    self.horizontalTabBar.contentView = [[UIVisualEffectView alloc] initWithEffect:[UIBlurEffect effectWithStyle:UIBlurEffectStyleProminent]];
}

- (TBTabBarControllerTabBarPlacement)preferredTabBarPlacementForViewSize:(CGSize)size {

    // Show the vertical tab bar whenever the device orientation is landscape

    return size.width >= size.height ? TBTabBarControllerTabBarPlacementLeading : TBTabBarControllerTabBarPlacementBottom;
}

@end

File: SwiftExample/TabBarController.swift
//
//  TabBarController.swift
//  SwiftExample
//
//  Created by Timur Ganiev on 09.12.2020.
//  Copyright © 2020-2023 Timur Ganiev. All rights reserved.
//

import UIKit
import TBTabBarController

final class TabBarController: TBTabBarController {

    // MARK: Overrides

    override func viewDidLoad() {
        
        super.viewDidLoad()

        // Make the bottom tab bar translucent
        
        horizontalTabBar.backgroundColor = .clear
        horizontalTabBar.contentView = UIVisualEffectView(effect: UIBlurEffect(style: .prominent))
    }

    override func preferredTabBarPlacement(
        forViewSize size: CGSize
    ) -> TBTabBarControllerTabBarPlacement {
        
        // Show the vertical tab bar whenever the device orientation is landscape
        
        return size.width >= size.height ? .leading : .bottom
    }
}
